### ISSUE TITLE
Generate Workflow Yaml

### ISSUE THREAD

**Issue by zachaller** [argoproj-labs/hera-workflows#28]
Is it possible to generate the workflow yaml instead of submitting the workflow?

**Comment by flaviuvadan** [comment]
Yes! Clients have access to the `workflow` field on a `Workflow` object, which stores the actual Argo SDK representation of the workflow:
```python
from hera.v1.task import Task
from hera.v1.workflow import Workflow
from hera.v1.workflow_service import WorkflowService

def foo():
  print(42)

ws = WorkflowService(...)
w = Workflow(...)
t = Task(...)
w.add(foo)

print(w.workflow.to_dict())
```

This prints out the dictionary representation of the workflow, which can be dumped to YAML via the `yaml` package.

**Comment by flaviuvadan** [comment]
Does that achieve what you need @zachaller? 🙂 Wondering if we can close this issue.

**Comment by Amitg1** [comment]
question, 
Can Hera generate Argo YAML spec? (not JSON).
I was thinking using it combined with kubeflow

**Comment by flaviuvadan** [comment]
> question, Can Hera generate Argo YAML spec? (not JSON). I was thinking using it combined with kubeflow

Hi @Amitg1! I believe you can `yaml.dump(workflow.spec)`!
